sw/device/silicon_creator/rom_ext/manifest_def.c

```diff
@@ -0,0 +1,20 @@
+// Copyright lowRISC contributors.
```

Comment:
From our offline discussion:
- I suggest moving this file to //sw/device/silicon_creator/lib
- Create a bazel build rule in that subdirectory:
  ```
  dual_cc_library(
    name = "manifest_def",
    srcs = dual_inputs(device=["manifest_def.c"]),
    hdrs = dual_inputs(shared=["manifest_def.h"]),
  )
  ```
- Eliminate the bare_metal_manifest_def.c.
- Use `//sw/device/silicon_creator/lib:manifest_def` in the rom_ext and bare_metal `deps`.

Reply:
This was massively helpful, thank you!